src/App/Application.cpp

```diff
@@ -35,6 +35,7 @@
 # endif
 # include <boost/program_options.hpp>
 # include <boost/date_time/posix_time/posix_time.hpp>
+# include <boost/scope_exit.hpp>

```

Comment:
This probably needs to be explicitly added to `PreCompiled.h` -- I don't know if there's a transitive include chain that pulls it in.

Reply:
Could you please elaborate on this? No one is including this cpp file, right?
I which regard is boost/scope_exit different from boost/date_time in the line above?
(please forgive me if I'm lacking some C++ 101 here)

Reply:
You have it in the right places, but since it's in the precompiled ifdef block, it won't be included on windows when using precompiled headers.

What chennes is saying is that you probably need to put it in this file too: https://github.com/FreeCAD/FreeCAD/blob/main/src/App/PreCompiled.h

Reply:
Yes, as @hyarion says -- compilation currently fails if using PCH because `boost/scope_exit.h` is missing from `src/App/PreCompiled.h`.

Reply:
I see. Thanks.